### PR TITLE
feat(rome_service): add `extends` functionality

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes https://github.com/rome/tools/issues/4038

This PR introduces breaking changes, mainly how the CLI handles broken configurations.

Before, a broken configuration would have resulted in warnings and Rome applying its defaults for the broken sections. I realised that while this is "cool", this isn't ideal for users because this would result in the following:
- false positives while running the linter
- incorrect formatting, with a configuration that isn't the one of the users

That's a breaking change.

Plus, now the `rome check` command shows diagnostics that belong to the formatter too.

## Test Plan

Added some test cases

## Changelog

- [x] The PR requires a changelog line

## Documentation

- [x] The PR requires documentation
- [x] I will create a new PR to update the documentation

### CODE REVIEW

File: crates/rome_cli/src/configuration.rs [hunk 1]
```diff
@@ -2,28 +2,53 @@ use std::path::PathBuf;
...
 impl LoadedConfiguration {
-    pub fn consume(self) -> (Configuration, Vec<rome_diagnostics::Error>, Option<PathBuf>) {
-        let path = self.path;
-        let (configuration, diagnostics) = self.deserialized.consume();
-        (configuration, diagnostics, path)
+    ///
+    pub fn apply_extends(&mut self, fs: &DynRef<dyn FileSystem>) -> Result<(), CliDiagnostic> {
+        let Some(extends) = &self.configuration.extends  else {return Ok(())};
+        let mut configuration_list = vec![];
+        let mut diagnostics = vec![];
```
Comment (denbezrukov) [inline crates/rome_cli/src/configuration.rs:23]: Do we need these temp variables?
Is it possible to use `self.diagnostics.extend` and `self.configuration.merge_with` directly in `for ... in`?

Comment (ematipico) [reply crates/rome_cli/src/configuration.rs]: @denbezrukov what do you think now?

Comment (denbezrukov) [reply crates/rome_cli/src/configuration.rs]: Oh, that borrow checker 😸 

I think that it's fine:)

**Comment by alex-grover** [comment]
@ematipico thanks for this feature, can I suggest adding it to the docs?

**Pull request merged**